xarray/util/generate_aggregations.py

```diff
@@ -508,11 +542,12 @@ def generate_code(self, method, has_keep_attrs):
             "\n" + 12 * " " + "keep_attrs=keep_attrs," if has_keep_attrs else ""
         )
         return f"""\
-        return self.reduce(
+        out = self.reduce(
```

Comment:
is this needed?

Reply:
I thought this looked nice and it was clear there was tweak to fix cumsum/cumprod:
```python
            out = self.reduce(
                duck_array_ops.cumsum,
                dim=dim,
                skipna=skipna,
                keep_attrs=keep_attrs,
                **kwargs,
            )
            return out.assign_coords(self._obj.coords)
```
Then for consistency and readability I followed that pattern on the others.